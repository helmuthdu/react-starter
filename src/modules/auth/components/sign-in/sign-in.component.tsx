import React from 'react';

type Props = {
  onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  ref: React.RefObject<HTMLInputElement>;
  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
  onClick: (evt: React.MouseEvent) => void;
  name: string;
};

export const SignIn = (props: Props) => (
  <form onSubmit={props.onSubmit}>
    <input ref={props.ref} type="text" placeholder="Username" onChange={props.onChange} required />
    <input type="password" placeholder="Password" required />
    <button type="submit" onClick={props.onClick}>
      Login
    </button>
    <p>current user: {props.name}</p>
  </form>
);
