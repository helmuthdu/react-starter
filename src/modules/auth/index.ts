import * as storesModule from './stores';

export { routes } from './routes';

export type State = Readonly<{
  auth: storesModule.auth.State;
}>;

export const stores = Object.values(storesModule);
