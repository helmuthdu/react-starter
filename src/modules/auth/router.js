import React from 'react';
import Loadable from 'react-loadable';
import DefaultLayoutRoute from './layouts/DefaultLayout';

const SignIn = Loadable({
  loader: () => import('./routes/SignIn/SignIn'),
  loading: () => null
});

export const AUTH_ROUTES = Object.freeze({
  SIGN_IN: '/sign-in'
});

export const authRoutes = [
  <DefaultLayoutRoute key={AUTH_ROUTES.SIGN_IN} path={AUTH_ROUTES.SIGN_IN} component={SignIn} />
];

export default authRoutes;
