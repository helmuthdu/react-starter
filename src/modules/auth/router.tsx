import React, { lazy } from 'react';
import DefaultLayoutRoute from './layouts/default.layout';

const SignIn = lazy(() => import('./routes/sign-in/sign-in.route'));

export const AUTH_ROUTES = Object.freeze({
  SIGN_IN: '/sign-in'
});

export const authRoutes = [
  <DefaultLayoutRoute key={AUTH_ROUTES.SIGN_IN} path={AUTH_ROUTES.SIGN_IN} component={SignIn} />
];

export default authRoutes;
