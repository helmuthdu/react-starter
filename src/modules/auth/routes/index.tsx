import React, { lazy } from 'react';
import DefaultLayoutRoute from '../layouts/default.layout';

const SignIn = lazy(() => import('./sign-in/sign-in.route'));

export const AUTH_ROUTES = Object.freeze({
  SIGN_IN: '/sign-in'
});

export const routes = [<DefaultLayoutRoute key={AUTH_ROUTES.SIGN_IN} path={AUTH_ROUTES.SIGN_IN} component={SignIn} />];

export default routes;
