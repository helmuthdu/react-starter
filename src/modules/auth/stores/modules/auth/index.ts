export * from './actions';
export * from './getters';
export * from './reducer';
export * from './sagas';
export * from './state';
export * from './types';

export const name = 'auth';
