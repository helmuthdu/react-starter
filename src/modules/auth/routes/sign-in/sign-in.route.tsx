import { push } from 'connected-react-router';
import React, { Component, HTMLAttributes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, compose, Dispatch } from 'redux';
import { AppState } from '../../../../app';
import { createSearchInputFromObservable } from '../../../../helpers';
import { SignIn } from '../../components/sign-in/sign-in.component';
import { auth } from '../../stores';
import { reduxForm } from 'redux-form';
import { Subject } from 'rxjs';

type StateProps = Readonly<{
  name: string;
}>;

type DispatchProps = auth.Actions & {
  linkTo: () => void;
};

type OwnProps = HTMLAttributes<HTMLFormElement>;

export type Props = StateProps & DispatchProps & OwnProps;

type State = Readonly<{
  inputValue: Subject<string>;
}>;

export class SignInRoute extends Component<Props, State> {
  public state = {
    inputValue: new Subject<string>()
  };

  public componentDidMount() {
    this.props.actionGetUser();
    createSearchInputFromObservable(this.state.inputValue, {}).subscribe((value: any) => {
      console.log('ON_CHANGE_WITH_OBSERVABLE: ', value);
    });
  }

  public render() {
    return (
      <SignIn
        onSubmit={e => e.preventDefault()}
        onChange={this.handleChange}
        onClick={this.handleClick}
        name={this.props.name}
      />
    );
  }

  private handleClick = (evt: React.MouseEvent) => {
    evt.preventDefault();
  };

  private handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
    this.state.inputValue.next(evt.currentTarget.value);
  };
}

const mapStateToProps = (state: AppState): StateProps => ({ name: state.auth.name });

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return bindActionCreators(
    {
      linkTo: () => push(`/`),
      ...auth.actions
    },
    dispatch
  );
};

const enhance = compose<React.ComponentClass<OwnProps>>(
  reduxForm({ form: 'signIn' }),
  connect<StateProps, DispatchProps, OwnProps, AppState>(
    mapStateToProps,
    mapDispatchToProps
  )
);

export default enhance(SignInRoute);
