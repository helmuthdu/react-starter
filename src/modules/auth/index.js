import routes from './router';
import * as stores from './store';

export default { routes, stores: Object.values(stores) };
