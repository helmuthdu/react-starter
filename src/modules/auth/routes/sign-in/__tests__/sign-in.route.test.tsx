import { shallow } from 'enzyme';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import { initialState } from '../../../store/modules/auth';
import { Props, SignInRoute } from '../sign-in.route';

describe('Route -> SignIn component', () => {
  const props: Props = {
    actionGetUser: jest.fn(),
    actionLogin: jest.fn(),
    actionLogout: jest.fn(),
    linkTo: jest.fn(),
    name: 'john doe'
  };

  const state = {
    dispatch: jest.fn(),
    store: configureMockStore()({ auth: initialState })
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<SignInRoute {...props} {...state} />);
    expect(wrapper).toMatchSnapshot();
  });
});
