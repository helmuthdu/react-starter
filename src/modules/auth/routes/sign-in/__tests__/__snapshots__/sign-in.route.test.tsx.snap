// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route -> SignIn component should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    placeholder="Username"
    required={true}
    type="text"
  />
  <input
    placeholder="Password"
    required={true}
    type="password"
  />
  <button
    type="submit"
  >
    Login
  </button>
  <p>
    current user: 
    john doe
  </p>
</form>
`;
